pkg/route/generator/generate.go

```diff
@@ -77,8 +78,14 @@ func (RouteGenerator) Generate(genericParams map[string]interface{}) (runtime.Ob
 
 	portString := params["target-port"]
 	if len(portString) > 0 {
+		var targetPort intstr.IntOrString
+		if port, err := strconv.Atoi(portString); err == nil {
+			targetPort = intstr.FromInt(port)
+		} else {
+			targetPort = intstr.FromString(portString)
+		}
 		route.Spec.Port = &api.RoutePort{
-			TargetPort: intstr.FromString(portString),
+			TargetPort: targetPort,
```

Comment:
> Actually since everything is clientside I am going to refactor the oc status part of the code to not use the conversion but decide based on the type. WDYT?

ugh, actually I just remembered why I couldn't do this in the first place. There may be existing routes (these coming from `oc expose` :) ) that use a route port with a string type but an int value...